reporter/html.js

```diff
@@ -795,7 +816,7 @@ QUnit.testDone( function( details ) {
 	assertList = testItem.getElementsByTagName( "ol" )[ 0 ];
 
 	good = details.passed;
-	bad = details.failed;
+	bad = details.failedUnexpectedly !== 0 || (details.todo && details.passedUnexpectedly === details.total);
```

Comment:
I think there's an edge case here where `details.total` is 0... not that we currently report that, but the second clause would be more robust as something like `details.passedUnexpectedly > 0 && details.passedUnexpectedly === details.total`.